library/network/src/modules/SuSEFirewall.rb

```diff
@@ -805,11 +830,67 @@ def Write
       true
     end
 
+    # Function returns if the interface is in zone.
+    #
+    # @param [String] interface
+    # @param	string firewall zone
+    # @return	[Boolean] is in zone
+    #
+    # @example IsInterfaceInZone ("eth-id-01:11:DA:9C:8A:2F", "INT") -> false
+    def IsInterfaceInZone(interface, zone)
+      interfaces = @SETTINGS[zone][:interfaces]
+      interfaces.include?(interface)
+    end
+
+    # Function returns the firewall zone of interface, nil if no zone includes
+    # the interface. Firewalld does not allow an interface to be in more than
+    # one zone, so no error detection for this case is needed.
+    #
+    # @param string interface
+    # @return string zone, or nil
+    def GetZoneOfInterface(interface)
+      Builtins.foreach(GetKnownFirewallZones()) do |zone|
```

Comment:
[`Builtins.foreach`](http://www.rubydoc.info/github/yast/yast-ruby-bindings/Yast/Builtins.foreach) is a relict from the YCP era.
Generally speaking, YCP library calls (Builtins.*, Ops.*) are designed to carry on in case they receive `nil`. We kept these during the mass conversion from Ruby to YCP since it was impossible to reason about nill-ness of values during the conversion. But here, since we know `GetKnownFirewallZones()` is never `nil`, we can simply call Ruby `each`